src/lib/hash.h

```diff
@@ -77,16 +78,42 @@ void pgp_calc_mdc_hash(
 
 const char *pgp_show_hash_alg(uint8_t);
 
-/* -----------------------------------------------------------------------------
- * @brief   Returns output size of an digest algorithm
+/* @brief   Returns output size of an digest algorithm
  *
  * @param   [in] alg
  * @param   [out] output_length: must be not NULL
  *
  * @return  true if provided algorithm is supported and it's size was
  *          correctly retrieved, otherwise false
- *
--------------------------------------------------------------------------------- */
+ **/
 bool pgp_digest_length(pgp_hash_alg_t alg, size_t *output_length);
 
+/* @brief Add hash for the corresponding algorithm to the list
+ * @param hashes non-NULL pointer to the list structure
+ * @param alg hash algorithm
+ * @return true if hash was added successfully or already exists in the list.
+ *         false will be returned if memory allocation failed, or alg is not supported, or
+ *         on other error
+ **/
+bool pgp_hash_list_add(list *hashes, pgp_hash_alg_t alg);
+
+/* @brief Get hash structure for the corresponding algorithm
+ * @param hashes List of pgp_hash_t structures
+ * @param alg Hash algorithm
+ * @return pointer to the pgp_hash_t structure or NULL if there is no one in the list
+ **/
+pgp_hash_t *pgp_hash_list_get(list hashes, pgp_hash_alg_t alg);
```

Comment:
It would be better to return const pointer here

Reply:
Agree.